app/code/Magento/Sales/Model/Order/Shipment/Track.php

```diff
@@ -415,5 +444,17 @@ public function setExtensionAttributes(\Magento\Sales\Api\Data\ShipmentTrackExte
         return $this->_setExtensionAttributes($extensionAttributes);
     }
 
+    /**
+     * Get Track Url for Track Item and return only if valid, absolute URI
+     *
+     * @return string|bool
+     */
+    public function getValidTrackUrl()
```

Comment:
It's not considered a good practice to add any methods to the DataInterface implementation other then getters and setters for the predefined properties.

Additionally, adding method only to interface implementation would force it's clients to always rely on this specific implementation and not the interface.